app_dart/lib/src/request_handling/cache_request_handler.dart

```diff
@@ -30,32 +30,29 @@ class CacheRequestHandler<T extends Body> extends RequestHandler<T> {
   /// [RequestHandler] to fallback on for cache misses.
   final RequestHandler<T> delegate;
 
-  final Cache<Uint8List> cache;
+  final CacheService cache;
 
   /// The time to live for the response stored in the cache.
   final Duration ttl;
 
+  @visibleForTesting
+  static const String responseSubcacheName = 'response';
+
   /// Services a cached request.
   @override
   Future<T> get() async {
-    final Cache<Uint8List> responseCache =
-        cache.withPrefix(await config.redisResponseSubcache);
-
     final String responseKey = '${request.uri.path}:${request.uri.query}';
-    final Uint8List cachedResponse = await responseCache[responseKey].get();
-
-    if (cachedResponse != null) {
-      return Body.forStream(Stream<Uint8List>.value(cachedResponse));
-    } else {
-      final Body body = await delegate.get();
-      final List<int> rawBytes = await body
-          .serialize()
-          .expand<int>((Uint8List chunk) => chunk)
-          .toList();
-      final Uint8List bytes = Uint8List.fromList(rawBytes);
-      await responseCache[responseKey].set(bytes, ttl);
-
-      return body;
-    }
+    final Uint8List cachedResponse = await cache.get(
+        responseSubcacheName, responseKey,
+        createFn: () => getBodyBytesFromDelegate(delegate), ttl: ttl);
+
+    return Body.forStream(Stream<Uint8List>.value(cachedResponse));
+  }
+
+  Future<Uint8List> getBodyBytesFromDelegate(RequestHandler<T> delegate) async {
+    final Body body = await delegate.get();
+    final List<int> rawBytes =
+        await body.serialize().expand<int>((Uint8List chunk) => chunk).toList();
+    return Uint8List.fromList(rawBytes);
   }
```

Comment:
I don't understand why converting body to bytes isn't trivial. Is it due to the body has a different encoding?

Reply:
This isn't trivial because we can only create a Uint8List from a list of ints.

`Uint8List value = Uint8List.fromList(await body.serialize().toList());`
Type error: `The argument type 'List<Uint8List>' can't be assigned to the parameter type 'List<int>'.dart(argument_type_not_assignable)`

We discussed this more in https://github.com/flutter/cocoon/pull/484, specifically https://github.com/flutter/cocoon/pull/484#discussion_r338807711 after some offline discussion.

Reply:
Non-trivial stuff deserves explanation in a comment.

Reply:
Updated with the explaination, and added documentation for the function.